com.unity.cinemachine/Editor/Upgrader/CinemachineUpgrader.cs

```diff
@@ -642,24 +644,78 @@ void Initialize(List<PlayableDirector> playableDirectors)
                         }
                     }
                 }
+                
+                // local function
+                static void LoadApiMapping()
+                {
+                    if (s_ApiUpgradeMaps != null)
+                        return;
+
+                    var filename = ScriptableObjectUtility.CinemachineRelativeInstallPath + 
+                        "/Editor/EditorResources/ApiUpgradeMapping.json";
+                    var json = System.IO.File.ReadAllText(filename);
+                    s_ApiUpgradeMaps = Newtonsoft.Json.JsonConvert
+                        .DeserializeObject<Dictionary<Type, Dictionary<string, Tuple<string, Type>>>>(json);
+                }
             }
+
+            static readonly Dictionary<Type, Type> k_ClassUpgradeMaps = new()
```

Comment:
I was trying to move specific class knowledge out of upgrader and into the upgraderForCM3 module.  That's why it got refactored to expose class lists instead of having them hardcoded here.  Can we try to stick to that paradigm, so we can keep this module as abstract as possible?  I'm thinking of future upgrade jobs.

Reply:
Sounds good. I'll move the dictionaries (cm2-cm3 specifics) there.